examples/structured_data/movielens_recommendations_transformers.py

```diff
@@ -0,0 +1,608 @@
+"""
+Title: A recommendation system with a transformer
+Author: [Khalid Salama](https://www.linkedin.com/in/khalid-salama-24403144/)
+Date created: 2020/12/30
+Last modified: 2020/12/30
+Description: Using the Behavior Sequence Transformer model for on Movielens dataset.
+"""
+
+"""
+## Introduction
+
+This example demonstrates the [Behavior Sequence Transformer (BST)](https://arxiv.org/abs/1905.06874)
+model, by Qiwei Chen et al., using the [Movielens dataset](https://grouplens.org/datasets/movielens/).
+The BST model leverages the sequential behaviour of the users in watching and rating movies,
+as well as user profile and movie features, to predict the rating of the user to a target movie.
+
+More precisely, the BST model aims to predict the rating of a target movie by accepting
+the following inputs:
+1. A fixed-length *sequence* of `movie_ids` watched by a user.
+2. A fixed-length *sequence* of the `ratings` for the movies watched by a user.
+3. A set of user features, including `user_id`, `sex`, `occupation`, and `age_group`.
```

Comment:
Since you highlight "sequence" above, might as well highlight "set" here and below.

Reply:
Done